lib/rubycritic/generators/console_report.rb

```diff
@@ -0,0 +1,52 @@
+require "colorize"
+
+module Rubycritic
+  module Generator
+    class ConsoleReport
+      def initialize(analysed_modules)
+        @analysed_modules = analysed_modules
+      end
+
+      def generate_report
+        reports = @analysed_modules.sort.map do |analysed_module|
+          generate_module_report analysed_module
+        end
+        puts reports.join("\n")
+      end
+
+      private
+
+      def generate_module_report(mod)
+        report = <<-EOF
+#{mod.name.send(mod.rating.color)}:
+    Rating:       #{mod.rating.to_s.send(mod.rating.color)}
+    Churn:        #{mod.churn}
+    Complexity:   #{mod.complexity}
+    Duplication:  #{mod.duplication}
+    Smells:       #{mod.smells.count}
+        EOF
+        smell_reports = mod.smells.map do |smell|
+          generate_smell_report smell
+        end
+        if smell_reports.any?
+          report << "\n"
+          report << smell_reports.join("\n")
+        end
+        report
+      end
+
+      def generate_smell_report(smell)
+        smell_line = smell.to_s
+        report = <<-EOF
+     * #{smell_line}
+        EOF
+        smell.locations.each do |location|
+          report << <<-EOF
+       - #{location}
+          EOF
```

Comment:
All those multiple heredocs make this method very hard to read. Additionally, I believe this method does way to much - we should split this up into small, maintainable methods.